packages/mixin-cache-as-bitmap/test/index.js

```diff
@@ -41,18 +42,69 @@ describe('PIXI.DisplayObject#cacheAsBitmap', function ()
 
         try
         {
-            renderer = new Renderer();
+            renderer = new Renderer({ width: 50, height: 50 });
             renderer.filter.push(par, par.filters);
+
+            const srcExpected = new Rectangle(5, 15, 10, 11);
+            const destExpected = new Rectangle(0, 0, 10, 11);
+
+            let src = renderer.renderTexture.sourceFrame;
+            let dest = renderer.renderTexture.destinationFrame;
+
+            expect(src.toString()).to.equal(srcExpected.toString());
+            expect(dest.toString()).to.equal(destExpected.toString());
+
+            obj.render(renderer);
+
+            src = renderer.renderTexture.sourceFrame;
+            dest = renderer.renderTexture.destinationFrame;
+
+            expect(obj._cacheData.sprite.width).to.equal(10);
+            expect(obj._cacheData.sprite.height).to.equal(11);
+            expect(src.toString()).to.equal(srcExpected.toString());
+            expect(dest.toString()).to.equal(destExpected.toString());
+        }
+        finally
+        {
+            renderer.destroy();
+        }
+    });
+
+    it('should respect projection', function ()
+    {
+        const par = new Container();
+        const obj = new Container();
+        let renderer = null;
+
+        par.filters = [new Filter()];
+        par.addChild(obj);
+
+        obj.position.set(5, 15);
+        obj.updateTransform();
+        obj.cacheAsBitmap = true;
+        obj._calculateBounds = function ()
+        {
+            this._bounds.clear();
+            this._bounds.addFrame(this.transform, 0, 0, 10, 11);
+        };
+
+        try
+        {
+            const srcExpected = new Rectangle(5, 15, 10, 11);
+            const destExpected = new Rectangle(20, 20, 10, 11);
+
+            renderer = new Renderer({ width: 50, height: 50 });
+            renderer.renderTexture.bind(null, srcExpected, destExpected);
+
             obj.render(renderer);
 
-            const frame = renderer.renderTexture.sourceFrame;
+            const src = renderer.renderTexture.sourceFrame;
+            const dest = renderer.renderTexture.destinationFrame;
 
             expect(obj._cacheData.sprite.width).to.equal(10);
             expect(obj._cacheData.sprite.height).to.equal(11);
-            expect(frame.x).to.equal(5);
-            expect(frame.y).to.equal(15);
-            expect(frame.width).to.equal(10);
-            expect(frame.height).to.equal(11);
+            expect(src.toString()).to.equal(srcExpected.toString());
```

Comment:
Interesting use case for toString!

Reply:
Well, you added it, and we have no equals, so I use it :)